skyportal/handlers/api/annotation.py

```diff
@@ -232,90 +232,94 @@ def post(self, associated_resource_type, resource_id):
 
         group_ids = data.pop('group_ids', None)
         if not group_ids:
-            groups = self.current_user.accessible_groups
-        else:
-            try:
-                groups = Group.get_if_accessible_by(
-                    group_ids, self.current_user, raise_if_none=True
-                )
-            except AccessError:
-                return self.error('Could not find any accessible groups.', status=403)
+            group_ids = [g.id for g in self.current_user.accessible_groups]
 
         if not isinstance(annotation_data, Mapping):
             return self.error(
                 "Invalid data: the annotation data must be an object with at least one {key: value} pair"
             )
 
-        author = self.associated_user_object
-
-        if associated_resource_type.lower() == "sources":
-            obj_id = resource_id
-            data['obj_id'] = obj_id
-            schema = Annotation.__schema__(exclude=["author_id"])
-            try:
-                schema.load(data)
-            except ValidationError as e:
+        with self.Session() as session:
+            author = self.associated_user_object
+            groups = session.scalars(
+                Group.select(self.current_user).where(Group.id.in_(group_ids))
```

Comment:
Why is author assigned above but not used here?

Reply:
It is used in the different if..else cases below.